### PR TITLE
git-find-repos: init at 2.1.0

### PR DESCRIPTION
**Pull request by yajo**
@moduon MT-1075

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/git-find-repos/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "acroz";
+    repo = "git-find-repos";
+    rev = version;
+    sha256 = "sha256-4TuZlt6XH4//DBHPuIMl/i3Tp6Uft62dGCTAuZ2rseE=";
+  };
+
+  build-system = [setuptools-scm];
```
Comment (momeemt) [inline pkgs/development/python-modules/git-find-repos/default.nix:19]:
```suggestion
  build-system = [ setuptools-scm ];
```

It's better to format with `nixfmt-rfc-style` that compliants with RFC 0166 (ref: https://github.com/NixOS/rfcs/pull/166).

File: pkgs/development/python-modules/git-find-repos/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "A simple CLI tool for finding git repositories.";
+    homepage = "https://github.com/acroz/git-find-repos";
+    license = licenses.mit;
+    maintainers = with maintainers; [yajo];
```
Comment (momeemt) [inline pkgs/development/python-modules/git-find-repos/default.nix:25]:
```suggestion
    maintainers = with maintainers; [ yajo ];
```

ditto

File: pkgs/development/python-modules/git-find-repos/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  };
```
Comment (momeemt) [inline pkgs/development/python-modules/git-find-repos/default.nix:26]:
```suggestion
  meta = {
    description = "A simple CLI tool for finding git repositories.";
    homepage = "https://github.com/acroz/git-find-repos";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ yajo ];
  };
```

The use of `with lib;` is discouraged. ref: https://github.com/NixOS/nixpkgs/pull/293767
You can put `lib.licenses.mit` in `meta.license`, and others too.

**Review by momeemt — CHANGES_REQUESTED** [review]
Thank you for your contributions! I have left some comments. I would be grateful if you could confirm these.

File: pkgs/development/python-modules/git-find-repos/default.nix (hunk 1 above)
Comment (yajo) [reply pkgs/development/python-modules/git-find-repos/default.nix]: Thanks! Honestly I'd expect some out-of-the-box pre-commit or treefmt integration at this point if nixpkgs cares about formatting.

Done manually nevertheless.

**Review by yajo — COMMENTED** [review]
Thanks, all attended.

**Review by momeemt — APPROVED** [review]
LGTM :)

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 310098` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.git-find-repos</li>
    <li>python311Packages.git-find-repos.dist</li>
    <li>python312Packages.git-find-repos</li>
    <li>python312Packages.git-find-repos.dist</li>
  </ul>
</details>

LGTM, just a nit

File: pkgs/development/python-modules/git-find-repos/default.nix [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+  build-system = [ setuptools-scm ];
+
+  meta = {
+    description = "A simple CLI tool for finding git repositories";
+    homepage = "https://github.com/acroz/git-find-repos";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.yajo ];
+  };
```
[the remark this answers is not in the document]
Comment (yajo) [reply pkgs/development/python-modules/git-find-repos/default.nix:26]: AFAICS [it shouldn't be necessary](https://nix.dev/manual/nix/2.22/command-ref/new-cli/nix3-run#description) if the binary name equals the derivation's `pname`, and that's the case here. Is there any other additional reason to set `mainProgram` here?

Thanks!

[the remark this answers is not in the document]
Comment (pbsds) [reply pkgs/development/python-modules/git-find-repos/default.nix:26]: We're trying to deprecate that behaviour in nixpkgs :)

[the remark this answers is not in the document]
Comment (yajo) [reply pkgs/development/python-modules/git-find-repos/default.nix]: Weird! Done, anyway. Thanks :)

**Comment by pbsds** [comment]
@ofborg build python3Packages.git-find-repos python3Packages.git-find-repos.passthru.tests

**Pull request merged**